modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -2542,4 +2553,13 @@ private void destroyTableStorageOnRecoveryBusy(CatalogTableDescriptor tableDescr
 
         engine.dropMvTable(tableDescriptor.id());
     }
+
+    private synchronized ScheduledExecutorService streamerFlushExecutor() {
+        if (streamerFlushExecutor == null) {
+            streamerFlushExecutor = Executors.newSingleThreadScheduledExecutor(
```

Comment:
Are we protected from first call to this method arriving after `close()` is closed (leaving the executor never shut down)?

Reply:
Added busy lock check